h2o-py/h2o/explanation/_explain.py

```diff
@@ -1304,44 +1309,48 @@ def ice_plot(
 
             factor_map = _factor_mapper(NumpyFrame(frame[column]).from_factor_to_num(column))
 
-        deciles = [int(round(frame.nrow * dec / 10)) for dec in range(11)]
-        colors = plt.get_cmap(colormap, 11)(list(range(11)))
         plt.figure(figsize=figsize)
-        for i, index in enumerate(deciles):
+        if display_mode == "both" or display_mode == "ice":
+            deciles = [int(round(frame.nrow * dec / 10)) for dec in range(11)]
+            colors = plt.get_cmap(colormap, 11)(list(range(11)))
+            for i, index in enumerate(deciles):
+                tmp = NumpyFrame(
+                    model.partial_plot(
+                        frame,
+                        cols=[column],
+                        plot=False,
+                        row_index=index,
+                        targets=target,
+                        nbins=20 if not is_factor else 1 + frame[column].nlevels()[0]
```

Comment:
It works like it will give you max nbins points from which you interpolate the line for numeric variable and it will give you nbins points (paint point per category) for categorical variable.

Currently, this is a copy/paste of original implementation - I've moved the original block to different place. I will have to work with this parameter in https://h2oai.atlassian.net/browse/PUBDEV-8494 - part 'specify number of points to simulate for each original observation (default of 100)'  since control over this parameter is required there. 

Some optimization for big cardinality variables is a good idea, maybe to plot only `some_threshold` top frequent values WDYT @tomasfryda ? But if it is supposed to be configurable in PUBDEV-8494, then I think it could be done like ` nbins=100 if not is_factor else 1 + frame[column].nlevels()[0]` in default case and `num_points_to_simulate` top frequent values when threshold will be specified by the user.

Reply:
@koniecsveta It would definitely be useful also I think it might be useful to be able to specify a list of categories to plot - sometimes the more important behavior can be on the less frequent categories (e.g., to find out if the model is negatively affecting some minority).

Reply:
ok, I will link this to https://h2oai.atlassian.net/browse/PUBDEV-8494 as a part of analysis

Reply:
@koniecsveta I was more interested in how continuous variables are handled. Take an example e.g. from [Numpy](https://numpy.org/doc/stable/reference/generated/numpy.histogram_bin_edges.html#numpy.histogram_bin_edges), they provide wide method selection for optimal bin size estimation. One of the most frequently used is Freedman-Diaconis rule. In Autoviz, we however used a different method chosen by Leland, but I cannot remember the name from the top of my head. I can however share the formula, if needed. The main issue with fixing the number of bins is that you may accidentally hide anomalies (outliers, inliers, ..) in the data, which may be important for the result.

Reply:
@johnygomez I've checked and in h2o-3 user can either define their own split points like this:

[pyunit_pubdev_5706_usersplits_pdp.py](https://github.com/h2oai/h2o-3/blob/master/h2o-py/tests/testdir_misc/pyunit_pubdev_5706_usersplits_pdp.py)

or they are calculated evenly just like this: 

```
if (v.isInt() && (v.max() - v.min() + 1) < _nbins) {
  actualbins = (int) (v.max() - v.min() + 1);
}
colVals = _add_missing_na ? new double[actualbins+1] : new double[actualbins];
double delta = (v.max() - v.min()) / (actualbins - 1);
if (actualbins == 1) delta = 0;
for (int j = 0; j < colVals.length; ++j) {
  colVals[j] = v.min() + j * delta;
}
```

[hex.PartialDependence#extractColValues](https://github.com/h2oai/h2o-3/blob/master/h2o-core/src/main/java/hex/PartialDependence.java#L195)

but I haven't found any bin size estimation method(s), we seem to not provide such thing.

Reply:
Ok no worries, that's just my 2 cents and a suggestion for potential improvement :+1: